### PR TITLE
Clusterio v17 migration

### LINKED ISSUE #4: Migrate to clusterio v17
The plugin does not currently work with the latest version of clusterio after the message and configuration refactor. Update the plugin to use the new message format.

### PR DESCRIPTION
**Pull request by Danielv123**
Should do the job unless I missed something critical, will convert to normal PR once I have gotten home and had some time to test it with the game installed.

resolves #4

### CODE REVIEW

**Comment by Danielv123** [comment]
This PR is now ready for review. From my initial testing stuff works as it should, with the same issues as it has had in the past (sometimes I need to restart a server after first configuring edges and some stuff, will have to look into that later).

I am not quite sure about the casing - messages now use camelCase, but the lua code uses snake_case. Some of this currently bleeds a bit over as we pass objects directly from Lua to network messages. I am fine with it the way it is now but did consider changing it to camelCase all the way through to the lua side, although that would break our conventions.

**Comment by Danielv123** [comment]
There is also an issue where this.logger in in the instance plugin does not output to my terminal. It does however output to the terminal in the web interface. I do not believe this has anything to do with the plugin code, so haven't looked that closely into it yet.

File: module/edge_logic.lua [hunk 1]
```diff
@@ -16,9 +16,9 @@ local belt_type_to_loader_type = {
 }
 
 
-local function vec2_sadd(a, s)
-	return {a[1] + s, a[2] + s}
-end
+-- local function vec2_sadd(a, s)
+-- 	return {a[1] + s, a[2] + s}
+-- end
```
Comment (Hornwitser) [inline module/edge_logic.lua:21]: Do we need to uncomment unused functions? They are still useful when changing the vector math for the edge logic.

File: module/edge_logic.lua [hunk 2]
```diff
@@ -32,9 +32,9 @@ local function vec2_smul(a, s)
 	return {a[1] * s, a[2] * s}
 end
 
-local function vec2_mul(a, b)
-	return {a[1] * b[1], a[2] * b[2]}
-end
+-- local function vec2_mul(a, b)
+-- 	return {a[1] * b[1], a[2] * b[2]}
+-- end
```
Comment (Hornwitser) [inline module/edge_logic.lua:37]: Same here.

File: module/edge_logic.lua [hunk 3]
```diff
@@ -497,7 +497,7 @@ function edge_transports.set_active_edges(json)
 
 		if not edge.active then
 			if edge.linked_belts then
-				for offset, link in pairs(edge.linked_belts) do
+				for _, link in pairs(edge.linked_belts) do
```
Comment (Hornwitser) [inline module/edge_logic.lua:500]: Does replacing the name of the key here with an underscore make the code more readable? I get that offset is unused, but by calling it offset it's understandable what the key is, wheras now I have to go look at other code dealing with linked_belts to figure out what the key is.

**Review by Hornwitser — COMMENTED** [review]
Looks fairly good. I have some questions about the luacheck changes in the Lua files and the following missed migration in controller.js and instance.js:

The signature for `onInstanceConfigChanged` has changed, it's now `async onInstanceConfigFieldChanged(instance: InstanceInfo, field: string, curr: unknown, prev: unknown)` on a controller plugin and the same with the `instance` argument left out on a host plugin. So instead of code like `if (group.name === "edge_transports" && field === "internal")` you now do `if (field === "edge_transports.internal")`.

**Pull request merged**